interfaces/builtin/hardware_udev_control.go

```diff
@@ -0,0 +1,54 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const hardwareUdevControlSummary = `allows control over the hardware udev daemon`
+
+const hardwareUdevControlBaseDeclarationSlots = `
```

Comment:
this looks fine for the slot, not sure if we want to make the plug super-privileged or not, I'm leaning towards making it super privileged since I assume the udevadm command can be used to inject new events a la `udevadm trigger --action=add ...` but maybe that doesn't actually allow one to do enough things to meaningfully achieve sandbox escape

Reply:
one can totally hose the whole machine with this interface, for example `udevadm control --stop-exec-queue` will stop udev daemon from processing any events, meaning any hotplugs or networking changes will not be acted upon by udev or anything listening on udev sockets (i.e. udevadm monitor).

Given that it is an open door to break all hardware access, imho it should be super-priviledged.

I don't think it allows escaping sandbox directly, but it does allow to start random systemd units. For example, `trigger --action=add` may start systemd units if any of the matching devices have SYSTEMD_WANTS rules and those systemd units exists.

Reply:
ack, thanks for specifying that, in this case can you make the plug side super-privileged with this:

```go
const hardwareUdevControlBaseDeclarationPlugs = `
  hardware-udev-control:
    allow-installation: false
    deny-auto-connection: true
`
```

and then add it in the common:

```go
	registerIface(&commonInterface{
		name:                  "hardware-udev-control",
 		summary:               hardwareUdevControlSummary,
		implicitOnCore:        true,
		implicitOnClassic:     true,
		baseDeclarationSlots:  hardwareUdevControlBaseDeclarationSlots,
		baseDeclarationPlugs:  hardwareUdevControlBaseDeclarationPlugs,
		connectedPlugAppArmor: hardwareUdevControlConnectedPlugAppArmor,
	})

```